numpy/ma/tests/test_core.py

```diff
@@ -1808,6 +1808,11 @@ def test_default_fill_value_structured(self):
         assert_equal(f1, expected)
         assert_equal(f2, expected)
 
+    def test_default_fill_value_void(self):
+        dt = np.dtype([('v', 'V7')])
```

Comment:
OK, I see they are slightly different tests. But the `dt = np.dtype('V7')` case is also broken in master and fixed by this PR.

For instance this fails in master:
```python
dt = np.dtype('V7')
np.array(np.ma.default_fill_value(dt), dt)
```

I would just add one extra test of some form for plain `V7` type, and we can merge. I think this PR is pretty obviously a good fix, so I don't mean to engage in too much bikeshedding for the tests here.